sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -844,6 +858,11 @@ public PartitionState apply(TopicPartition tp) {
           partitionStates.get(i).nextOffset = ckptMark.getNextOffset();
         }
       }
+
+      messagesConsumedPerSplit = Metrics.counter("io.splits", source.id + ".messagesConsumed");
+      bytesConsumedPerSplit = Metrics.counter("io.splits", source.id + ".bytesConsumed");
+      backlogBytesPerSplit = Metrics.gauge("io.splits", source.id + ".backlog.bytes");
+      backlogMessagesPerSplit = Metrics.gauge("io.splits", source.id + ".backlog.messages");
```

Comment:
Is it a performance concern? Even we keep couple of atomic longs ourselves, the cost might be same.